File: Api/src/main/java/net/binggl/mydms/features/shared/JsonDateSerializer.java
package net.binggl.mydms.features.shared;

import java.io.IOException;
import java.util.Date;

import org.joda.time.DateTime;
import org.joda.time.format.DateTimeFormatter;
import org.joda.time.format.ISODateTimeFormat;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

public class JsonDateSerializer extends JsonSerializer<Date> {

    private static DateTimeFormatter formatter = ISODateTimeFormat.dateTime();

    @Override
    public void serialize(Date value, JsonGenerator gen, SerializerProvider arg2)
            throws IOException, JsonProcessingException {
    	
        gen.writeString(formatter.print(new DateTime(value)));
    }

}

File: Api/src/main/kotlin/net/binggl/mydms/shared/util/JsonDateSerializer.kt
package net.binggl.mydms.shared.util

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import java.io.IOException
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class JsonDateSerializer : JsonSerializer<LocalDateTime>() {

    @Throws(IOException::class, JsonProcessingException::class)
    override fun serialize(value: LocalDateTime, gen: JsonGenerator, arg2: SerializerProvider) {
        gen.writeString(formatter.format(value))
    }

    companion object {
        private val formatter = DateTimeFormatter.ISO_DATE_TIME
    }

}